Consumers/StripeMessageConsumer.cs

```diff
@@ -0,0 +1,180 @@
+                    }
+                }
+                catch (Exception e)
+                {
+                    _logger.Log($"Error processing message from callbackQueue: {e}", LogLevels.Error);
+                }
+
+                await Task.Delay(2000);
+            }
+        }
+
+        async void InvoicePaymentSucceeded(Event stripeEvent, Message message)
+        {
+            var invoice = stripeEvent.Data.Object as Invoice;
+            var subscription = new SubscriptionModel
+            {
+                SubscriptionId = invoice!.SubscriptionId,
+                CustomerId = invoice.CustomerId,
+                Email = invoice.CustomerEmail,
+                UserStatus = invoice.Total == 0 ? UserStatus.FreeTrial : UserStatus.Subscribed,
+                InvoiceUrl = invoice.HostedInvoiceUrl,
+                StartDate = invoice.Lines.Data[0].Period.Start,
+                EndDate = invoice.Lines.Data[0].Period.End
+            };
+
+            await _subscriptionService.CreateSubscription(subscription);
+            await DeleteMessage(message.ReceiptHandle);
+        }
+
+        async void UpdateUserStatus(Event stripeEvent, Message message)
+        {
+            var updatedSubscription = stripeEvent.Data.Object as Subscription;
+            var response = await _subscriptionService.GetSubscriptionById(updatedSubscription!.Id);
+            var newUserStatus = UserStatus.Unknown;
+
+            var currentSubscription = response.Data;
+            if (currentSubscription == null) { return; }
+
+            // Determine new UserStatus based on the updated subscription's status
+            switch (updatedSubscription.Status) {
+                case "trialing":
+                    newUserStatus = UserStatus.FreeTrial;
+                    break;
+                case "paused":
+                    newUserStatus = UserStatus.FreeTrialEnded;
+                    break;
+                case "active":
+                    newUserStatus = UserStatus.Subscribed;
+                    break;
+                case "canceled":
+                    newUserStatus = currentSubscription.UserStatus == UserStatus.FreeTrial ? UserStatus.FreeTrialEnded : UserStatus.SubscriptionEnded;
+                    break;
+                case "unpaid":
+                    newUserStatus = UserStatus.NotPaid;
+                    break;
+                case "past_due":
+                    newUserStatus = UserStatus.NotPaid;
+                    break;
+                default:
+                    newUserStatus = UserStatus.Unknown;
+                    break;
+            }
+
+            // Create recurring job if UserStatus moves to FreeTrial or Subscribed
+            if (newUserStatus == UserStatus.FreeTrial || newUserStatus == UserStatus.Subscribed)
+            {
+                var family = _familyService.GetByEmail(currentSubscription.Email).Data;
+                _jobService.CreateorUpdateJob(family!.Id);
```

Comment:
Two separate families having the same email shouldn't occur because of the login in the CreateProfile function in the Index Controller

Reply:
ahh I see, With the way it works now someone could create a login for an email that is not theirs since we don't require the email verification. It does create two different users in auth0 as well

Reply:
I meant logic instead of login. And I thought auth0 did email verification?